### ISSUE TITLE
Possible quirk/exploit of example "reflection" enchantment.

### ISSUE THREAD

**Issue by CrimsonDawn45** [CrimsonDawn45/Fabric-Shield-Lib#105]
Since Arrow entities don't delete themselves when they strike a shield with reflection there is a chance it will hover in the air repeatedly striking the shield with the reflection enchantment infinitely damaging the mob.

I'm mostly putting this here because my other mod which adds a bunch of shields uses the reflection enchantment so I remember how to fix this later. Also we might want to implement a fix to the test enchantment as that enchantment is meant to also act as a built in tutorial of sorts on how to add a shield enchantment.

**Comment by CrimsonDawn45** [comment]
Since we have that enchantment as an example on the wiki anyone who uses the tutorial will also have this bug...

**Comment by CringeStar** [comment]
I don't think this is an issue, since I was playing around with the shield with the reflection enchantment, and when skeletons shot me, Tthis bug didn't _seem_ to occur. But I'm not sure, I'll test it again

**Comment by CringeStar** [comment]
I tested this yesterday, this bug doesn't occur. One other thing that I did notice was that if you were blocking, and were damaged from behind, where the shield doesn't blo k damage, the reflection would still occur and the entity would still be damaged.

**Comment by CrimsonDawn45** [comment]
Huh, also it was one of those very very specific bugs where the arrow had to hit the shield at a perfect angle

**Comment by CrimsonDawn45** [comment]
idk it was mostly just a fluke, but still running into that happening while I was playing the mod in survival to test the shields was very strange.

**Comment by CrimsonDawn45** [comment]
> I tested this yesterday, this bug doesn't occur. One other thing that I did notice was that if you were blocking, and were damaged from behind, where the shield doesn't block damage, the reflection would still occur and the entity would still be damaged.

But this, this should maybe be looked into because if the enchantments can't tell whether the blocking of damage was successful than that's a problem. I imagine we might have to make our event invocation do the math to figure out the angles & add in a Boolean of true or false if the hit was on the actual shield.

**Comment by CringeStar** [comment]
"The shield directionally blocks all attacks coming from within the FOV of the direction the wielder is facing, providing a full hemisphere of coverage to them. If the wielder faces straight up, they are likely to miss their blocks." -Minecraft wiki. 

I've tested this, and I'm pretty sure its not accurate, as FOV doesn't matter. I did narrow down the degree that the shield block to be between 110-180 degrees.
